Makefile.work

```diff
@@ -184,7 +190,7 @@ ifeq ($(SONIC_CONFIG_USE_NATIVE_DOCKERD_FOR_BUILD), y)
     DOCKER_RUN += -v /var/run/docker.sock:/var/run/docker.sock
 endif
 
-ifeq ($(MULTIARCH_QEMU_ENVIRON), y)
+ifneq ($(filter y, $(MULTIARCH_QEMU_ENVIRON) $(CROSS_BUILD_ENVIRON)),)
```

Comment:
Hi @gregshpit,

Why should the slave container be using `/var/run/march/docker.sock, /var/run/march/docker.pid` instead of `/var/run/docker.sock & /var/run/docker.pid`  as it seems you were pulling the slave docker `FROM debian:buster` as opposed to the qemu compilation which uses `FROM multiarch/debian-debootstrap:arm64-buster`

My build server (amd64), which i've traditionally used compile for amd64 targets doesn't have `/var/run/march/docker.sock & /var/run/march/docker.pid` and is failing when trying to build any DOCKER_TARGETS using cross compilation. 

But it is working, when i remove this condition `ifneq ($(filter y, $(MULTIARCH_QEMU_ENVIRON) $(CROSS_BUILD_ENVIRON)),)` and restore the original condition i.e `ifeq ($(MULTIARCH_QEMU_ENVIRON), y)`

Reply:
/var/run/march/docker.sock docker daemon should start automatically. But there is a bug in Makefile.work. To make the automation work in cross-compilation mode please replace line 297 in Makefile.work:
-ifeq ($(MULTIARCH_QEMU_ENVIRON), y)

with 
+ifneq ($(filter y, $(MULTIARCH_QEMU_ENVIRON) $(CROSS_BUILD_ENVIRON)),)

This should start the march docker daemon automatically.
Let me know if it helped.

Reply:
In my case though, it didn't work with the new condition.  i.e using this command:

`docker run --rm=true --privileged --init -v <build_dir>/sonic-buildimage:/sonic -w /sonic -e http_proxy= -e https_proxy= -e no_proxy= -it -v /var/cache/sonic/artifacts:/dpkg_cache:rw -v /var/run/march/docker.sock:/var/run/docker.sock -v /var/run/march/docker.pid:/var/run/docker.pid -v /var/run/march/docker:/var/run/docker -v /var/lib/march/docker:/var/lib/docker`

probably because i didn't have the /var/run/march/docker.sock setup. I'll raise the query in the community. thank you